packages/ai-jsx/src/react/index.ts

```diff
@@ -92,14 +95,24 @@ export function useAI(children: LLMx.Node) {
 /**
  * A JSX component that allows AI.jsx elements to be used in a React component tree.
  */
-export function jsx({ children }: { children: LLMx.Node }, context?: any | LLMx.ComponentContext) {
+export function jsx(
+  {
+    children,
+    onStreamStart,
+    onStreamEnd,
+    loading = '',
+  }: { children: LLMx.Node; onStreamStart?: () => void; onStreamEnd?: () => void; loading?: React.ReactNode },
+  context?: any | LLMx.ComponentContext
+) {
```

Comment:
@petersalas I think it'll be useful for UIs to know when a generation is in progress. Lmk what you think / if you have an alternate suggestion for how to do it.

The `loading` function is the moral equivalent of `Suspense'`s `fallback`. I chose not to use that name because `fallback` to me implies `failed`, as opposed to `not-started`.

Reply:
I'm wondering if a) useAI should take the value as an "initialValue" and b) components that want to observe the lifecycle should use useAI directly. On first glance that feels more consistent with how other hooks/async operations are modeled in React. WDYT?

Reply:
I don't have a strong opinion now but I expect to develop one as we build more apps.

I think some usecases will definitely want to access `useAI` directly, and some (most?) of those usecases will want to observe the lifecycle.

But I also think a purely declarative approach (like what I did in this PR) can righteously benefit from observing the lifecycle as well.

But maybe I'll change my mind on all this later.

re: initial value – I don't think I'm fully following.